src/lib/server/hyrise_server.cpp

```diff
@@ -18,7 +19,8 @@ void HyriseServer::accept_next_connection() {
 
 void HyriseServer::start_session(boost::system::error_code error) {
   if (!error) {
-    std::make_shared<HyriseSession>(std::move(_socket), _io_service)->start();
+    auto connection = std::make_shared<ClientConnection>(std::move(_socket));
+    std::make_shared<HyriseSession>(_io_service, connection)->start();
```

Comment:
comment here that this needs to be shared_ptr because we call shared_from_this in start. otherwise a shared_ptr makes no sense